pkg/manager/internal/controllers/project_controller.go

```diff
@@ -108,12 +113,40 @@ func (r *ProjectReconciler) reconcileMembers(ctx context.Context, project *stora
 	return nil
 }
 
+// Project Namespace name is organization_name-separator-project_name
+func generateProjectNamespaceName(organizatioName, projectName string) string {
+	return fmt.Sprintf("%s%s%s", organizatioName, projectNamespaceNameSeparator, projectName)
+}
+
+// Project Namespace name is organization_name-separator-project_name
+func deconstructProjectNamespaceName(projectNamespaceName string) (string, string, error) {
+	split := strings.Split(projectNamespaceName, projectNamespaceNameSeparator)
+	if len(split) != 2 {
+		return "", "", fmt.Errorf("not a valid project namespace name to deconstruct: %s", projectNamespaceName)
+	}
+
+	return split[0], split[1], nil
+}
+
 func (r *ProjectReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	enqueuer := owner.EnqueueRequestForOwner(&storagev1alpha1.Project{}, mgr.GetScheme())
 
-	// TODO watch rolebindings in project namespaces
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&storagev1alpha1.Project{}).
 		Watches(&source.Kind{Type: &corev1.Namespace{}}, enqueuer).
+		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}}, &handler.EnqueueRequestsFromMapFunc{
+			ToRequests: handler.ToRequestsFunc(func(object handler.MapObject) []reconcile.Request {
+				org, pro, err := deconstructProjectNamespaceName(object.Meta.GetNamespace())
+				if err != nil {
+					return []reconcile.Request{}
```

Comment:
Anyway, this piece of code should:

* fetch the namespace in which `rbacv1.RoleBinding` resides
* reconcile for namespace owner; something like:

```
	enqueuer := owner.EnqueueRequestForOwner(&storagev1alpha1.Project{}, mgr.GetScheme()).(*handler.EnqueueRequestsFromMapFunc)  // owner.* could either export the MapFunc, or return the type versus the interface; either way changing the API is beneficial. 

	return ctrl.NewControllerManagedBy(mgr).
		For(&storagev1alpha1.Project{}).
		Watches(&source.Kind{Type: &corev1.Namespace{}}, enqueuer).
		Watches(&source.Kind{Type: &rbacv1.RoleBinding{}}, &handler.EnqueueRequestsFromMapFunc{
			ToRequests: handler.ToRequestsFunc(func(object handler.MapObject) []reconcile.Request {
				ctx := context.Background()
				ns := &corev1.Namespace{}
				if err := r.Client.Get(ctx, types.NamespacedName{Name: object.Meta.GetNamespace()}, ns); err != nil {
					panic(err)
				}
				return enqueuer.ToRequests.Map(handler.MapObject{
					Meta:   ns,
					Object: ns,
				})
			}),
		}).
		Complete(r)
}
```

Reply:
Yeah, I agree with Neven here that we should avoid parsing Name/Namespace from string.

Reply:
Thanks, sounds like a better idea, ill do that

Reply:
keep in mind you'll have to open PR in utils repo for owner API to change in some way. Casting to `(*...)` isn't nice in the long run.

Reply:
it's fine for this PR, but not longterm.

Reply:
ok, ill make a note to change it in util repo